File: Problem_Solving/Algorithms/Warmup/8_Mini_Max_Sum/Solution.js
'use strict';

process.stdin.resume();
process.stdin.setEncoding('utf-8');

let inputString = '';
let currentLine = 0;

process.stdin.on('data', inputStdin => {
    inputString += inputStdin;
});

process.stdin.on('end', _ => {
    inputString = inputString.replace(/\s*$/, '')
        .split('\n')
        .map(str => str.replace(/\s*$/, ''));

    main();
});

function readLine() {
    return inputString[currentLine++];
}

function miniMaxSum(arr) {
    const arrLen=arr.length;
    let sum=0;
    let minVal=Number.MAX_SAFE_INTEGER;
    let maxVal=Number.MIN_SAFE_INTEGER;

    for(let i=0;i<arrLen;i++) {
        sum+=arr[i];
        if(arr[i]>maxVal)
            maxVal=arr[i];
        if(arr[i]<minVal)
            minVal=arr[i];
    }

    console.log(`${sum-maxVal} ${sum-minVal}`);
}

function main() {
    const arr = readLine().split(' ').map(arrTemp => parseInt(arrTemp, 10));

    miniMaxSum(arr);
}

File: Problem_Solving/Algorithms/Warmup/8_Mini_Max_Sum/Solution.cs
using System.CodeDom.Compiler;
using System.Collections.Generic;
using System.Collections;
using System.ComponentModel;
using System.Diagnostics.CodeAnalysis;
using System.Globalization;
using System.IO;
using System.Linq;
using System.Reflection;
using System.Runtime.Serialization;
using System.Text.RegularExpressions;
using System.Text;
using System;

class Solution {

    static void miniMaxSum(int[] arr) {
        long sum=0;
        long minVal=long.MaxValue;
        long maxVal=long.MinValue;

        foreach(int num in arr){
            sum+=num;
            if(num>maxVal)
                maxVal=num;
            if(num<minVal)
                minVal=num;
        }
        
        Console.Write((sum-maxVal) + " " + (sum-minVal));
    }

    static void Main(string[] args) {
        int[] arr = Array.ConvertAll(Console.ReadLine().Split(' '), arrTemp => Convert.ToInt32(arrTemp));
        miniMaxSum(arr);
    }
}
